lib/chef/resource/macos_pkg.rb

```diff
@@ -0,0 +1,111 @@
+#
+# Author:: Joshua Timberman (<jtimberman@chef.io>)
+# Author:: William Theaker (<william.theaker+chef@gusto.com>)
+# Copyright:: Copyright (c) Chef Software Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require_relative "../resource"
+
+class Chef
+  class Resource
+    class MacosPkg < Chef::Resource
```

Comment:
Ultimately, this should be using the `Chef::Resource::Package` pattern that other package providers use. (See [`lib/chef/resource/homebrew_package.rb](https://github.com/chef/chef/blob/main/lib/chef/resource/homebrew_package.rb)).

Reply:
Also look at [`lib/chef/provider/package/homebrew.rb`](https://github.com/chef/chef/blob/main/lib/chef/provider/package/homebrew.rb)

Reply:
macOS pkg files don't really have a concept of uninstalling. Would the `Chef::Resource::Package` pattern  still be useful when the only action in allowed_actions is :install?

Reply:
You can override them and mark them unsupported.

But we should support uninstall along these lines: https://gist.github.com/githubutilities/b5318d08a4b970d104f1

List the files, remove them, and then do the `--forget`.